internal/gmm/handler.go

```diff
@@ -125,19 +125,16 @@ func transport5GSMMessage(ue *context.AmfUe, anType models.AccessType,
 				}
 				ue.GmmLog.Warningf("Duplicated PDU session ID[%d]", pduSessionID)
 				smContext.SetDuplicatedPduSessionID(true)
-				response, errResponse, _, err := consumer.SendUpdateSmContextRequest(smContext, updateData, nil, nil)
+				response, _, _, err := consumer.SendUpdateSmContextRequest(smContext, updateData, nil, nil)
 				if err != nil {
+					smContext.SetDuplicatedPduSessionID(false)
```

Comment:
I think the duplicated attribute should be `true` if `SendUpdateSmContextRequest()` gets error.

Reply:
I think it is better to unset the duplicated attribute when `SendUpdateSmContextRequest()` gets error for the following reasons: 
* When the network initiates PDU session release (e.g. 4.3.4.2 1e in TS 23.502) for some reason after `SendUpdateSmContextRequest()` fails, if the attribute is true, AMF tries to create the new SMContext by invocation of Nsmf_PDUSession_SMContextStatusNotify. If the attribute is false, PDU session release completes and no new session is established.
* If `SendUpdateSmContextRequest()` gets error, AMF doesn't send response to UE. On the expiry of the timer T3580 UE retransmits the PDU Session Establishment Request, the attribute can be set again, so there is no problem.

Reply:
@mariakagi 
I got it, thanks for your explanation!

Reply:
@mariakagi 

It's better to do local release smContext when error or error response happened:
```
				if err != nil {
					ue.GmmLog.Errorf("SendUpdateSmContextRequest err: %v, local release SmContext[%d]",
						err, pduSessionID)
					ue.DeleteSmContext(pduSessionID)
					return err
				} else if response == nil {
					ue.GmmLog.Errorf("SendUpdateSmContextRequest rsp is nil, local release SmContext[%d]",
						pduSessionID)
					ue.DeleteSmContext(pduSessionID)
				} else if response != nil {

```

Reply:
@tim-ywliu 
Does `ue.DeleteSmContext(pduSessionID)` mean `ue.SmContextList.Delete(pduSessionID)` ?

Reply:
@mariakagi 

Yes.

Reply:
Your suggestion looks good.
But when `if err != nil`, `"SendUpdateSmContextRequest err: ..."` is output twice.

Reply:
@mariakagi 

Ya, the error log here can be removed.

Reply:
I made a modification to delete smContext and revise the text output to log.